File: Examples/test-suite/perl5/template_inherit_runme.pl
use strict;
use warnings;
use Test::More tests => 17;
BEGIN { use_ok('template_inherit') }
require_ok('template_inherit');

my $a = template_inherit::FooInt->new();
my $b = template_inherit::FooDouble->new();
my $c = template_inherit::BarInt->new();
my $d = template_inherit::BarDouble->new();
my $e = template_inherit::FooUInt->new();
my $f = template_inherit::BarUInt->new();

is($a->blah(), 'Foo');
is($b->blah(), 'Foo');
is($e->blah(), 'Foo');
is($c->blah(), 'Bar');
is($d->blah(), 'Bar');
is($f->blah(), 'Bar');
is($c->foomethod(), 'foomethod');
is($d->foomethod(), 'foomethod');
is($f->foomethod(), 'foomethod');
is(template_inherit::invoke_blah_int($a), 'Foo');
is(template_inherit::invoke_blah_int($c), 'Bar');
is(template_inherit::invoke_blah_double($b), 'Foo');
is(template_inherit::invoke_blah_double($d), 'Bar');
is(template_inherit::invoke_blah_uint($e), 'Foo');
is(template_inherit::invoke_blah_uint($f), 'Bar');

File: SWIG/Examples/test-suite/python/template_inherit_runme.py
from template_inherit import *
a = FooInt()
b = FooDouble()
c = BarInt()
d = BarDouble()

if a.blah() != "Foo":
    raise ValueError

if b.blah() != "Foo":
    raise ValueError

if c.blah() != "Bar":
    raise ValueError

if d.blah() != "Bar":
    raise ValueError

if c.foomethod() != "foomethod":
    raise ValueError

if d.foomethod() != "foomethod":
    raise ValueError

if invoke_blah_int(a) != "Foo":
    raise ValueError

if invoke_blah_int(c) != "Bar":
    raise ValueError

if invoke_blah_double(b) != "Foo":
    raise ValueError

if invoke_blah_double(d) != "Bar":
    raise ValueError

